flow-client/src/main/frontend/form/Validators.ts

```diff
@@ -71,15 +71,15 @@ abstract class NumberValidator<T> extends AbstractValidator<T> {
   }
 }
 
-export class IsNumber extends NumberValidator<number | undefined> {
+export class IsNumber extends NumberValidator<number | null | undefined> {
   optional: boolean;
   constructor(optional: boolean, attrs?: ValidatorAttributes) {
     super({ message: 'must be a number', ...attrs });
     this.optional = optional;
   }
 
-  validate(value: number | undefined) {
-    return (this.optional && value === undefined) || super.validate(value);
+  validate(value: number | null | undefined) {
+    return (this.optional && (value === null || value === undefined)) || super.validate(value);
```

Comment:
We can use the loose null check here:
```suggestion
    return (this.optional && value == null) || super.validate(value);
```

Reply:
cool, didn't know about this trick, thanks